packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts

```diff
@@ -453,10 +454,18 @@ export const usePickerValue = <
     onSelectedSectionsChange: handleFieldSelectedSectionsChange,
   };
 
+  const isValid = (testedValue: TValue) =>
+    validator({
+      adapter,
+      value: testedValue,
+      props: { ...props, value: testedValue },
+    }) === null;
```

Comment:
Or something similar to support range pickers.
```suggestion
  const isValid = (testedValue: TValue) => {
    const validationResponse = validator({
      adapter,
      value: testedValue,
      props: { ...props, value: testedValue },
    });
    return Array.isArray(testedValue)
      ? (validationResponse as any[]).every((v) => v === null)
      : validationResponse === null;
  };
```